ts/src/main/scala/ch/epfl/ts/component/Component.scala

```diff
@@ -51,6 +94,24 @@ final class ComponentBuilder(val system: ActorSystem) {
     instances = new ComponentRef(system.actorOf(props, name), props.clazz, name, this) :: instances
     instances.head
   }
+  
+  /**
+   * Gracefully stop all managed components.
+   * When all stops are successful, we clear the `instances` list.
+   * 
+   * @return A future which completes when *all* managed actors have shut down.
+   */
+  def shutdownManagedActors(timeout: FiniteDuration = 3 seconds): Future[Unit] = {
```

Comment:
This is a delicate point, and I think that the difference is:
- `start` / `stop` could be called several times (like restarting actors without actually killing them)
- `shutdown` really kills the actor

I will try to make it clear in the documentation.